### PR TITLE
Refactor the library to use magefile and better structure

### PR DESCRIPTION
**Pull request by fzerorubigd**
This is a proposal. I am open to any suggestion or objection. 

The structure is based on [this structure](https://github.com/golang-standards/project-layout) 
Also the package names are normally lowercase single word. and also the name cmd is mostly used as parent for main package
Another change is to move to [magefile](httpw://magefile.org) it is like make files, but with Go.

### CODE REVIEW

**Comment by fzerorubigd** [comment]
Since it can breaks things, I thing its a good idea to decide on #28 first and the continue with this one.

File: magefile.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"os"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+// Build build the wait4it binary. This build is static and is not using CGO
+func Build() error {
```
Comment (ph4r5h4d) [inline magefile.go:13]: Niiice, I didn't know about this possibilty. thank you.

**Comment by ph4r5h4d** [comment]
I agree with changing the versioning.
These changes are fine to me, I think it's now much better and GO way.
Thank you for the time you spent on making the project structure better.

**Comment by fzerorubigd** [comment]
so lets create a develop branch, and target this to develop branch for the next release

**Comment by ph4r5h4d** [comment]
If everything else is ok with you I can merge it.

**Comment by fzerorubigd** [comment]
I guess that's ok then.

**Pull request merged**